src/__tests__/app.integration.test.js

```diff
@@ -199,3 +226,127 @@ describe('POST /registration-requests/', () => {
     });
   });
 });
+
+describe('Ensure health record outbound XML is unchanged', () => {
+  // ============ COMMON PROPERTIES ============
+  // EHR Data
+  const ODS_CODE = "B85002";
+
+  // Database Models
+  const MessageFragment = ModelFactory.getByName(messageFragmentModel);
+  const RegistrationRequest = ModelFactory.getByName(registrationRequestModel);
+
+  // Responses
+  const DEFAULT_RESULT = {
+    hasFailed: false,
+    inProgress: false
+  };
+  // =================== END ===================
+
+  beforeAll(async () => {
+    await MessageFragment.truncate();
+    await RegistrationRequest.truncate();
+    await MessageFragment.sync({ force: true });
+    await RegistrationRequest.sync({ force: true });
+  });
+
+  afterAll(async () => {
+    await MessageFragment.sequelize.sync({ force: true });
+    await RegistrationRequest.sequelize.sync({ force: true });
+    await ModelFactory.sequelize.close();
+  });
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should verify that a small EHR core is unchanged by XML changes', async () => {
+    // given
+    const NHS_NUMBER = 9693796047;
+    const CONVERSATION_ID = "0005504B-C4D5-458A-83BD-3FA2CCAE650E";
+    const EHR_REQUEST_ID = "A4709C25-DD61-4FED-A9ED-E35AA464A7B3";
+    const ORIGINAL_UK06 = readFile('RCMR_IN030000UK06', 'equality-test', 'small-ehr', 'original');
+
+    // when
+    getEhrCoreFromRepo.mockResolvedValueOnce(Promise.resolve(ORIGINAL_UK06));
+    patientAndPracticeOdsCodesMatch.mockResolvedValueOnce(Promise.resolve(true));
+    sendCore.mockResolvedValueOnce(Promise.resolve(undefined));
+
+    const response = await transferOutEhrCore({
+      conversationId: CONVERSATION_ID,
+      nhsNumber: NHS_NUMBER,
+      odsCode: ODS_CODE,
+      ehrRequestId: EHR_REQUEST_ID
+    });
+
+    const MODIFIED_UK06 = sendCore.mock.calls[0][2];
+
+    // then
+    expect(validateMessageEquality(ORIGINAL_UK06, MODIFIED_UK06)).toBe(true);
+    expect(response).toEqual(DEFAULT_RESULT);
+  });
+
+  it('should verify that a fragment with no external attachments is unchanged by xml changes', async () => {
+    // given
+    const NHS_NUMBER = 9693643038;
+    const CONVERSATION_ID = "05E36C93-2DEF-4586-B842-127C534FB8B7";
+    const ORIGINAL_FRAGMENTS = {
+      ['DD92589A-B5B4-4492-AADD-51534821F07B']: readFile('COPC_IN000001UK01_01', 'equality-test', 'large-ehr-no-external-attachments', 'original'),
+      ['DE4A9436-FFA3-49B1-8180-5570510F0C11']: readFile('COPC_IN000001UK01_02', 'equality-test', 'large-ehr-no-external-attachments', 'original'),
+      ['62330782-1C8B-45CD-95E3-4FC624091C61']: readFile('COPC_IN000001UK01_03', 'equality-test', 'large-ehr-no-external-attachments', 'original'),
+      ['770C42DD-301B-4177-A78E-0E9E62F3FDA1']: readFile('COPC_IN000001UK01_04', 'equality-test', 'large-ehr-no-external-attachments', 'original')
```

Comment:
Lets change the keys from single element array to just a string?
e.g. `['DD92589A-B5B4-4492-AADD-51534821F07B']` to `'DD92589A-B5B4-4492-AADD-51534821F07B'`
In javascript it will convert all key value to a string so test would still pass, but using an array as a key isn't very good